File: testdata/jennies/builders/initialization_safeguards/JavaBuilder/initialization_safeguards/SomePanelBuilder.java
package initialization_safeguards;


public class SomePanelBuilder implements cog.Builder<SomePanel> {
    protected final SomePanel internal;
    
    public SomePanelBuilder() {
        this.internal = new SomePanel();
    }
    public SomePanelBuilder title(String title) {
        this.internal.title = title;
        return this;
    }
    
    public SomePanelBuilder showLegend(Boolean show) {
		if (this.internal.options == null) {
			this.internal.options = new initialization_safeguards.Options();
		}
		if (this.internal.options.legend == null) {
			this.internal.options.legend = new initialization_safeguards.LegendOptions();
		}
        this.internal.options.legend.show = show;
        return this;
    }
    public SomePanel build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/initialization_safeguards/PHPBuilder/src/InitializationSafeguards/SomePanelBuilder.php
<?php

namespace Grafana\Foundation\InitializationSafeguards;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\InitializationSafeguards\SomePanel>
 */
class SomePanelBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\InitializationSafeguards\SomePanel $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\InitializationSafeguards\SomePanel();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\InitializationSafeguards\SomePanel
     */
    public function build()
    {
        return $this->internal;
    }

    public function title(string $title): static
    {
        $this->internal->title = $title;
    
        return $this;
    }

    public function showLegend(bool $show): static
    {    
        if ($this->internal->options === null) {
            $this->internal->options = new \Grafana\Foundation\InitializationSafeguards\Options();
        }
        assert($this->internal->options instanceof \Grafana\Foundation\InitializationSafeguards\Options);
        $this->internal->options->legend->show = $show;
    
        return $this;
    }

}
